From b396862580c1cd2497179aa8ffd5615d6345fced Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:08 -0400
Subject: [block] blk-mq: fix leak of hctx->ctx_map

Message-id: <1402670238-13916-81-git-send-email-snitzer@redhat.com>
Patchwork-id: 83856
O-Subject: [RHEL7.1 PATCH 080/150] blk-mq: fix leak of hctx->ctx_map
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 1f9f07e917f43af420f8cb3afc2b3fa703cea6e9
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Tue May 27 08:34:45 2014 -0600

    blk-mq: fix leak of hctx->ctx_map

    hctx->ctx_map should have been freed inside blk_mq_free_queue().

    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 62082c5..dad22a9 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1846,6 +1846,7 @@ void blk_mq_free_queue(struct request_queue *q)
  queue_for_each_hw_ctx(q, hctx, i) {
   blk_mq_tag_idle(hctx);
   kfree(hctx->ctxs);
+  blk_mq_free_bitmap(&hctx->ctx_map);
   blk_mq_unregister_cpu_notifier(&hctx->cpu_notifier);
   if (q->mq_ops->exit_hctx)
    q->mq_ops->exit_hctx(hctx, i);
-- 
1.7.1